src/intelligence_layer/connectors/data/data.py

```diff
@@ -0,0 +1,245 @@
+        """Create a new dataset in a repository.
+
+        Args:
+            repository_id: Repository ID
+            dataset: :DatasetCreate object
+
+        Returns:
+            :class:`DataDataset` new object
+        """
+        url = urljoin(
+            self._base_data_platform_url,
+            f"api/v1/repositories/{repository_id}/datasets",
+        )
+        body = {
+            "sourceData": dataset.source_data,
+            "labels": ",".join(dataset.labels),
+            "name": dataset.name,
+            "totalDatapoints": dataset.total_datapoints,
+            "metadata": json.dumps(dataset.metadata) if dataset.metadata else None,
+        }
+        response = self._do_request(
+            "POST",
+            url,
+            files={k: v for k, v in body.items() if v not in [None, ""]},
+        )
+        return DataDataset(**response.json())
+
+    def list_datasets(
+        self, repository_id: str, page: int = 0, size: int = 20
+    ) -> list[DataDataset]:
+        """List all the datasets in a repository.
+
+        Args:
+            repository_id: Repository ID
+            page: Page number. Defaults to 0
+            size: Number of items per page. Defaults to 20
+
+        Returns:
+            List of :class:`DataDataset` from a given repository
+        """
+        url = urljoin(
+            self._base_data_platform_url,
+            f"api/v1/repositories/{repository_id}/datasets",
+        )
+        query = urlencode({"page": page, "size": size})
+        response = self._do_request("GET", f"{url}?{query}")
+        datasets = response.json()
+        return [DataDataset(**dataset) for dataset in datasets["datasets"]]
+
+    def get_dataset(self, repository_id: str, dataset_id: str) -> DataDataset:
+        """Get a dataset by ID.
+
+        Args:
+            repository_id: Repository ID
+            dataset_id: DataDataset ID
+
+        Returns:
+            :class:`DataDataset` object
+        """
+        url = urljoin(
+            self._base_data_platform_url,
+            f"api/v1/repositories/{repository_id}/datasets/{dataset_id}",
+        )
+        response = self._do_request("GET", url)
+        return DataDataset(**response.json())
+
+    def delete_dataset(self, repository_id: str, dataset_id: str) -> None:
+        """Delete a dataset by ID.
+
+        Args:
+            repository_id: Repository ID
+            dataset_id: DataDataset ID
+        """
+        url = urljoin(
+            self._base_data_platform_url,
+            f"api/v1/repositories/{repository_id}/datasets/{dataset_id}",
+        )
+        self._do_request("DELETE", url)
+
+    def stream_dataset(self, repository_id: str, dataset_id: str) -> Iterator[Any]:
+        """Stream the data points of a dataset.
+
+        Args:
+            repository_id: Repository ID
+            dataset_id: DataDataset ID
+
+        Returns:
+            :class Iterator[Any] of datapoints
```

Comment:
:class is broken here. It can just be "Iterator over the datapoints of the given dataset"